### DISCUSSION TITLE
Reading netcdf file

### DISCUSSION THREAD

**Discussion by carlos9917** [dtcenter/METplus#2525] [File I/O] [answered]
Hello,
I am trying to use a netcdf with grid_stat, but since it does not work I tried reading it first with plot_data_plane.
However, I get the same cryptic message:

DEBUG 1: Start plot_data_plane by nhd(17054) at 2024-03-26 09:40:48Z  cmd: /perm/nhd/MET/bin/plot_data_plane /perm/nhd/data_sample/MET_Norway_snow/daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc test.ps prob_nosnow_c -v 6 
ERROR  : 
ERROR  : yyerror() -> syntax error in file "/tmp/met_config_1930790_1"
ERROR  : 
ERROR  :    line   = 2
ERROR  : 
ERROR  :    column = 13
ERROR  : 
ERROR  :    text   = "(nul)"
ERROR  : 
ERROR  : 
ERROR  : (nul)
ERROR  : (nul)
ERROR  : 

The file is attached. Not sure what is wrong with the file? This is a downsampled version of a bigger file, but I get the same error from both. Not sure what the error means? The file is attached.
Thank you in advance for any help.
Carlos

[
[daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc.gz](https://github.com/dtcenter/METplus/files/14756244/daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc.gz)
](url)

**Comment by jprestop** [comment]
HI @carlos9917.  Carlos, thanks for explaining the problem you're having and for sending along the sample data to illustrate. I am able to replicate the problem you describe by running the `plot_data_plane` tool for MET version 11.1.0:

```
(base) jpresto$ /nrit/ral/met-11.1.0/bin/plot_data_plane daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc test.ps prob_snow_c -v 6
DEBUG 1: Start plot_data_plane by jpresto(7414) at 2024-03-26 18:49:45Z  cmd: /nrit/ral/met-11.1.0/bin/plot_data_plane daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc test.ps prob_nosnow_c -v 6 
ERROR  : 
ERROR  : yyerror() -> syntax error in file "/tmp/met_config_3731925_1"
ERROR  : 
ERROR  :    line   = 2
ERROR  : 
ERROR  :    column = 11
ERROR  : 
ERROR  :    text   = "(nul)"
ERROR  : 
ERROR  : 
ERROR  : (nul)
ERROR  : (nul)
ERROR  : 
```
Looking at the file you sent, I see that the following variable exists:

```
	float prob_snow_c(time, lat, lon) ;
		prob_snow_c:_FillValue = NaNf ;
		prob_snow_c:units = "some_unit" ;
```
However, we were using "prob_nosnow_c" and not "prob_snow_c".  While that is not the problem, it's still good to note and correct.  The current problem lies in the inputs to plot_data_plane.  The field string isn't specified correctly.  Looking at an example in our [documentation for plot_data_plane](https://met.readthedocs.io/en/main_v11.1/Users_Guide/plotting.html#plot-data-plane-usage), I've reformatted the command to the following (using `L0` as a generic level type):
```
(base) jpresto$ /nrit/ral/met-11.1.0/bin/plot_data_plane daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc test.ps 'name="prob_snow_c"; level="L0";' -v 6
```
That gets us farther, but we now have a new error:
```
DEBUG 1: Start plot_data_plane by jpresto(7414) at 2024-03-26 18:50:13Z  cmd: /nrit/ral/met-11.1.0/bin/plot_data_plane daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc test.ps name="prob_snow_c"; level="L0"; -v 6 
DEBUG 1: Opening data file: daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc
DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_None".
ERROR  : 
ERROR  : plot_data_plane -> file "daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc" not a valid data file
ERROR  : 
```
Perhaps using the full file instead of the downsampled version, you will not receive this error.  Please give it a try and let us know how it goes.  If you receive the same error, please send the full file to our ftp server using these [directions](https://github.com/dtcenter/METplus/discussions/954) and let us know once you've put the file there so we can grab it.

**Reply by jprestop** [reply, marked as answer]
@carlos9917 Just a follow up - I incorrectly specified the level.  For NetCDF files, the level should be specified in the following way:
`level="(0,*,*)";`.  Modifying the command for the correct specification of the level and adding `file_type = NETCDF_NCCF;` as an input produces the following results. For more information on specifying the level and file_type, please see our documentation in [this section](https://met.readthedocs.io/en/develop/Users_Guide/config_options.html#fcst).
```
(base) jpresto@seneca:/d1/projects/METplus/discussions/2525$ /nrit/ral/met-11.1.0/bin/plot_data_plane daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc test.ps 'name="prob_snow_c"; level="(0,*,*)"; file_type = NETCDF_NCCF;' -v 6
DEBUG 1: Start plot_data_plane by jpresto(7414) at 2024-03-26 20:24:06Z  cmd: /nrit/ral/met-11.1.0/bin/plot_data_plane daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc test.ps name="prob_snow_c"; level="(0,*,*)"; file_type = NETCDF_NCCF; -v 6 
DEBUG 1: Opening data file: daily-avhrr-sce-nhl_ease-50_201505011200_resampled.nc
DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_NcCF".
DEBUG 6: has_unsigned_attribute() -> time does not have _Unsigned attribute.
DEBUG 6: get_nc_data(NcVar *, double *) time data_size=1, is_unsigned_value: 0
DEBUG 5: add_to_unixtime() -->19700101_000000 plus 0 times 0 seconds = 19700101_000000
DEBUG 4: NcCfFile::open() -> could not extract init time from the "forecast_reference_time" variable.
DEBUG 4: NcCfFile::open() -> could not extract init time from file name.
DEBUG 4: 
DEBUG 4: Latitude/Longitude Grid Data:
DEBUG 4:      lat_ll: 55.9
DEBUG 4:      lon_ll: 109.7
DEBUG 4:   delta_lat: 0.1
DEBUG 4:   delta_lon: 0.1
DEBUG 4:        Nlat: 302
DEBUG 4:        Nlon: 1475
DEBUG 4: 
DEBUG 4: VarInfoFactory::new_var_info() -> created new VarInfo object of type "FileType_NcCF".
DEBUG 2: 
DEBUG 2: NcCfFile::getData(NcVar *, const LongArray &, DataPlane &) const -> data was flipped to north.
DEBUG 6: NcCfFile::getData(NcVar *, const LongArray &, DataPlane &) const -> took 0.020142 seconds
DEBUG 4: 
DEBUG 4: Grid information:
DEBUG 4:    Projection: Lat/Lon
DEBUG 4:    Nx: 1475
DEBUG 4:    Ny: 302
DEBUG 4:    lat_ll: 55.900
DEBUG 4:    lon_ll: 109.700
DEBUG 4:    delta_lat: 0.100
DEBUG 4:    delta_lon: 0.100
DEBUG 4:    wrapLon: false
DEBUG 4: 
DEBUG 4: Data plane information:
DEBUG 4:    plane min: 0
DEBUG 4:    plane max: 100
DEBUG 4:    valid time: 19700101_000000
DEBUG 4:    lead time: 000000
DEBUG 4:    init time: 19700101_000000
DEBUG 4:    accum time: 000000
DEBUG 4: 
DEBUG 1: Creating postscript file: test.ps
DEBUG 4: draw_map() -> Plotting map data file: /nrit/ral/met-11.1.0/share/met/map/country_data
DEBUG 4: draw_map() -> Plotting map data file: /nrit/ral/met-11.1.0/share/met/map/usa_state_data
DEBUG 4: draw_map() -> Plotting map data file: /nrit/ral/met-11.1.0/share/met/map/major_lakes_data
DEBUG 1: Finish plot_data_plane by jpresto(7414) at 2024-03-26 20:24:07Z
```
However, please note that the time value of 0 listed in the file is just interpreted as unixtime 0:
```
DEBUG 5: add_to_unixtime() -->19700101_000000 plus 0 times 0 seconds = 19700101_000000
```

**Reply by carlos9917** [reply]
Thank you very much for all the details! I really appreciate it.
That works. I will try it now with grid_stat
Best wishes
Carlos

**Reply by jprestop** [reply]
Great! Thanks so much for letting us know.  I'm going to go ahead and lock this discussion to prevent future posts here, as they sometimes get lost, if these are marked as Answered.  Please start a new discussion with any new issues.
